dashboards/create-metrics.py

```diff
@@ -0,0 +1,110 @@
+import argparse
+import json
+import logging
+from google.cloud import logging as gcloud_logging
+
+
+def main(service_account_path, pre_defined_metrics_json='metrics_template.json', google_project=None):
+    logging.basicConfig(level=logging.INFO)
+
+    # Instantiate a client from the service account JSON key path
+    logging_client = gcloud_logging.Client.from_service_account_json(service_account_path)
+
+    # Load pre-defined metrics JSON
+    with open(pre_defined_metrics_json, 'r') as f:
+        METRICS = json.load(f)
+
+    # List the current metrics
+    list_metrics(logging_client)
+
+    # Create the necessary metrics from the pre-defined list
+    if google_project:
+        # Check if we are using the right service account JSON key
+        assert logging_client.project == google_project
+    google_project = logging_client.project
+
+    for metric in METRICS:
+        check_and_create_metrics(gcloud_log_client=logging_client, metric_json=metric, google_project=google_project)
+
+    # List the current metrics again
+    list_metrics(logging_client)
+
+    # Double check
+    assert len(METRICS) == len(list(logging_client.list_metrics()))
+
+
+def list_metrics(gcloud_log_client):
+    """List all of the log-based metrics that is currently available on the cloud.
```

Comment:
What do you mean by listing all of the metrics "on the cloud"? Are you referring to the gcloud project?

Reply:
Sorry for the confusion! This means that list all of the remote metrics of the google_project.